.github/workflows/terraform.yaml

```diff
@@ -0,0 +1,156 @@
+#   state will be protected against inconsistencies by the lock provided by
+#   Terraform Cloud but the other tests will result in undefined behavior.
+# - If Terraform fails or is interrupted during deployment or destruction, the
+#   state might end up locked and subsequent runs will fail. In this case the
+#   state first needs to be unlocked, for instance from the Terraform Cloud UI.
+
+# Set these secrets to configure the CI run
+env:
+  VAST_PEERED_VPC_ID: "${{ secrets.VAST_PEERED_VPC_ID }}"
+  VAST_CIDR: "${{ secrets.VAST_CIDR }}"
+  VAST_AWS_REGION: "${{ secrets.VAST_AWS_REGION }}"
+  TF_ORGANIZATION: "${{ secrets.TF_ORGANIZATION }}"
+  TF_WORKSPACE_PREFIX: "${{ secrets.TF_WORKSPACE_PREFIX }}"
+  TF_API_TOKEN: ${{ secrets.TF_API_TOKEN }}
+  AWS_ACCESS_KEY_ID: "${{ secrets.AWS_ACCESS_KEY_ID }}"
+  AWS_SECRET_ACCESS_KEY: "${{ secrets.AWS_SECRET_ACCESS_KEY }}"
+  VASTCLOUD_NOTTY: 1
+
+jobs:
+  vast_on_aws:
+    name: "VAST ON AWS"
+    runs-on: ubuntu-20.04
+    defaults:
+      run:
+        working-directory: ./cloud/aws
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Terraform Format
+        run: terraform fmt -check -recursive -diff
+
+      - name: "Create config file"
+        if: ${{ !env.ACT }}
+        run: |
+          echo "vast_peered_vpc_id = $VAST_PEERED_VPC_ID" >> .env
+          echo "vast_cidr = $VAST_CIDR" >> .env
+          echo "vast_aws_region = $VAST_AWS_REGION" >> .env
+
+      - name: "Substitute backend with Terraform Cloud"
+        run: |
+          cat<<EOF > step-1/backend.tf
+          terraform {
+            cloud {
+              organization = "$TF_ORGANIZATION"
+              token        = "$TF_API_TOKEN"
+              workspaces {
+                name = "$TF_WORKSPACE_PREFIX-step-1"
+              }
+            }
+          }
+          EOF
+          cat<<EOF > step-2/backend.tf
+          terraform {
+            cloud {
+              organization = "$TF_ORGANIZATION"
+              token        = "$TF_API_TOKEN"
+              workspaces {
+                name = "$TF_WORKSPACE_PREFIX-step-2"
+              }
+            }
+          }
+          EOF
+
+      - name: Deploy
+        id: deploy
+        continue-on-error: true
+        run: ./vast-cloud deploy --auto-approve
+
+      - name: Retry deploy
+        if: steps.deploy.outcome=='failure'
+        run: |
+          echo "Deploy sometimes fails for an unexplained reason."
+          echo "Retrying after a few minutes usually works."
+          sleep 200
+          ./vast-cloud deploy --auto-approve
+
+      - name: Start and restart VAST server
+        run: |
+          echo "Run start-vast-server"
+          ./vast-cloud start-vast-server
+
+          echo "Run start-vast-server again"
+          ./vast-cloud start-vast-server 2> /dev/null \
+            && { echo "Starting server again should fail"; false; } \
+            || true
```

Comment:
Instead of this complicated Boolean logic, can we use an if condition here for readability? Same below this snippet.

Reply:
I wanted to avoid making it too verbose. But you are right, this syntax is cryptic (in fact, I think the statement you highlighted is wrong, because it does not fail when it should). 

Isn't there a better way to assert in bash? Another solution would be to add a `vast-cloud integgration-test` command that runs all these. This brings a lots of benefits:
- give the ability to run the tests locally
- take advantage of the python syntax to do the asserts
- the github action code remains simpler

Reply:
> Another solution would be to add a `vast-cloud integgration-test` command that runs all these

That sounds like a great idea. Local reproducability is always helpful